lib/tests/detectors/lpar_detector.cc

```diff
@@ -20,4 +20,20 @@ SCENARIO("Using the LPAR detector") {
             REQUIRE(res.get<int>("partition_number") == 16);
         }
     }
+
+    WHEN("Running outside of an LPAR but `lparstat` is available") {
```

Comment:
I'd change these to:

```
WHEN("Running on a non-AIX machine") {
   THEN("the result is not valid") {
   }
}

WHEN("Running on an AIX machine") {
    WHEN("lparstat is available") {
        THEN(...)
    }
    WHEN("lparstat is unavailable") {
        THEN(...)
    }
}
```

makes it clearer that these detectors only make sense for AIX machines.